### DISCUSSION TITLE
Can you use Remix with TypeScript 5.2?

### DISCUSSION THREAD

**Discussion by tmcw** [remix-run/remix#7615] [Q&A]
I'm a little confused by the scenario as it stands. Today I happened upon some code in my app that could _really_ benefit from TypeScript 5.2's new [using](https://www.totaltypescript.com/typescript-5-2-new-keyword-using) keyword, but after tinkering with an implementation, realized that eslint didn't know about `using` and would complain about it.

So, my first instinct was to upgrade our copy of `@typescript-eslint/eslint-plugin` to `~6` which should bring in new TypeScript. But then it gets a peerDependencies conflict with `@remix-run/eslint-config`, which wants it to still be 5. So, sure, I haven't upgraded to Remix 2.x yet, still on 1.19.3 - so maybe it's updated in 2.x, but checking latest says that it's still pinned to `^5.59.0` of typescript-eslint's plugin and parser.

So, long story short - can you run TypeScript 5.2 with Remix and have eslint working?

**Comment by kiliman** [comment]
My understanding is that the Remix team plans to remove `@remix-run/eslint-config` as part of the core set of packages and leave lint configs up to user stacks.

https://github.com/remix-run/remix/pull/7597

The PR will show you the recommended settings and will be included in the default Remix template. Just copy those to your project and you should be fine.

**Comment by HeathHopkins** [comment]
I'm attempting this too, and I get this error (remix dev server 2.1.0):

`app/routes/test.tsx:14:10: ERROR: Expected ";" but found "test2" [plugin browser-route-module]`

__app/routes/test.tsx__

```typescript
import type { LoaderFunctionArgs } from "@remix-run/node";
import { json } from "@remix-run/node";

export const loader = async ({ request }: LoaderFunctionArgs) => {
    const test = () => {
        return {
            name: "test",
            [Symbol.dispose]: () => {
                console.log("disposed")
            }
        }
    }

    using test2 = test() // line 14 where the error is reported

    console.log(`test2: ${test2.name}`)
    return json({
        name: test2.name
    })
}
```

I pasted the `.eslintrc.cjs` from the PR and I still receive the error.  Were either of you able to get `using` to work?

**Reply by kiliman** [reply]
The error is coming from `esbuild`. It doesn't currently support the `using` keyword.

**Reply by HeathHopkins** [reply]
Thanks
